src/polyfill/array-fill.js

```diff
@@ -0,0 +1,72 @@
+if (!Array.prototype.fill) {
+    Object.defineProperty(Array.prototype, 'fill', {
+        value: function(value) {
+  
+            // Steps 1-2.
+            if (this == null) {
+                throw new TypeError('this is null or not defined');
+            }
+    
+            var O = Object(this);
+    
+            // Steps 3-5.
+            var len = O.length >>> 0;
+    
+            // Steps 6-7.
+            var start = arguments[1];
+            var relativeStart = start >> 0;
+    
+            // Step 8.
+            var k = relativeStart < 0 ? Math.max(len + relativeStart, 0) : Math.min(relativeStart, len);
+    
+            // Steps 9-10.
+            var end = arguments[2];
+            var relativeEnd = end === undefined ? len : end >> 0;
+    
+            // Step 11.
+            var finalValue = relativeEnd < 0 ? Math.max(len + relativeEnd, 0) : Math.min(relativeEnd, len);
+    
+            // Step 12.
+            while (k < finalValue) {
+                O[k] = value;
+                k++;
+            }
+    
+            // Step 13.
+            return O;
+        }
+    });
+}
+
+// implementation for typed arrays
+if (!Int8Array.prototype.fill) {
+    Int8Array.prototype.fill = Array.prototype.fill;
```

Comment:
does this work correctly if `Array.fill` implementation was provided (not polyfilled)? might be worth splitting out the function and using it for all polyfills (including Array).

Reply:
Good point.

Reply:
I would not expect a problem with this? But we only have IE to test on and that is missing both. I tested on IE and this works, and is suggested polyfill (see links I added)

Reply:
I understand your test worked.

However this code suggests it's possible that `Array.fill` is provided by the browser while `Int8Array.fill` isn't. Under such a circumstance this code very well may fail.

It is simple enough to make this code not fail and I don't really understand why you don't do it?

Reply:
Also, currently a bit misleading because I would expect `array-fill.js` to just polyfill `Array#fill`. I'd put the rest in `typedarray-fill.js`.